lib/ui/setup/pair_page.dart

```diff
@@ -79,12 +79,14 @@ class PairPage extends HookWidget implements CobbleScreen {
 
     final _refreshDevicesBle = () {
       if (!scan.scanning) {
+        scan.devices.clear();
```

Comment:
Thinking about it again, I'd add a method on ScanCallbacks because it seems like there was an attempts to make state in ScanCallbacks immutable. ScanCallbacks always creates new state with `scanning` and `devices` field, so just clearing `devices` array is different than what is done otherwise in code. 

My suggestion is to call `context.read(scanProvider).onScanStarted();` or create similar method.

Reply:
I changed it to `context.refresh(scanProvider).onScanStarted();`. read was not working as expected

Reply:
@DreadBoy Should I mark this conversation resolved? Or is context.refresh not a preferable way to handle this?

Reply:
`refresh()` is fine too, it's similar to nuke option, where we dispose entire provider and create it anew. 😁 But if it works, it works, I wouldn't stress about it...

Reply:
I don't know how to approve/resolve the change request 🙁